orderbook/src/orderbook.rs

```diff
@@ -51,6 +53,25 @@ impl OrderBook {
             Err(RemoveOrderError::DoesNotExist)
         }
     }
+
+    // Clean up operations like removing expired orders.
+    pub async fn run_maintenance_forever(&self) -> ! {
+        loop {
+            tracing::debug!("running order book maintenance");
+            self.remove_expired_orders(SystemTime::now()).await;
+            tokio::time::delay_for(MAINTENANCE_INTERVAL).await;
+        }
+    }
+
+    async fn remove_expired_orders(&self, now: SystemTime) {
+        // TODO: use the timestamp from the most recent block instead?
+        let now = now
+            .duration_since(SystemTime::UNIX_EPOCH)
+            .expect("now earlier than epoch")
+            .as_secs();
+        let mut orders = self.orders.write().await;
+        orders.retain(|order| order.order_creation.valid_to as u64 > now);
```

Comment:
I would add here some buffer. If the tx needs to be settled within the next second, we should also remove it, as there is no chance that it will be settled.

Maybe this buffer should be: MAINTENANCE_INTERVAL

Reply:
Makes sense from the solver's point of view but I'm not sure about the orderbook. I feel the orderbook shouldnt' delete too early and let the solver's decide what they think their minimal handling time is.

Reply:
yeah, makes sense!